packages/opencensus-exporter-ocagent/test/test-ocagent.ts

```diff
@@ -0,0 +1,464 @@
+    const ROOT_SPAN_NAME = 'root-span';
+    const CHILD_SPAN_NAME = 'child-span';
+
+    // Create a rootSpan and one childSpan, then validate that those spans
+    // arrived at the server through the grpc stream.
+    tracing.tracer.startRootSpan(
+        {
+          name: ROOT_SPAN_NAME,
+          spanContext: {traceId: null, spanId: null, options: 0x1}
+        },
+        (rootSpan: RootSpan) => {
+          const childSpan = rootSpan.startChildSpan(CHILD_SPAN_NAME, '');
+
+          // When the stream is connected, we end both spans, which should
+          // trigger the spans to be sent to the agent.
+          server.once(MockAgentEvent.ExportStreamConnected, () => {
+            childSpan.end();
+            rootSpan.end();
+          });
+
+          // When the stream receives data, validate that the two spans we just
+          // ended are present.
+          server.once(
+              MockAgentEvent.ExportStreamMessageReceived,
+              (message: opencensus.proto.agent.trace.v1
+                   .ExportTraceServiceRequest) => {
+                let foundRootSpan = false;
+                let foundChildSpan = false;
+                message.spans.forEach(span => {
+                  switch (span.name.value) {
+                    case ROOT_SPAN_NAME: {
+                      foundRootSpan = true;
+                      break;
+                    }
+                    case CHILD_SPAN_NAME: {
+                      foundChildSpan = true;
+                      break;
+                    }
+                    default: { break; }
+                  }
+                });
+                assert.ok(foundRootSpan, 'could not find root span in message');
+                assert.ok(
+                    foundChildSpan, 'could not find child span in message');
+                done();
+              });
+        });
+  });
+
+  it('should receive probability sampler configuration changes from agent',
+     (done) => {
+       /**
+        * Have the server send a configuration change to use a probability
+        * sampler with the given probability, then validate that the current
+        * tracer sampler was updated correctly.
+        */
+       const runProbabilityTest = (testProbability: number): Promise<void> => {
+         server.sendConfigurationChangeProbability(testProbability);
+         return new Promise(resolve => {
+           setTimeout(() => {
+             const sampler = tracing.tracer.sampler.description;
+             // Validate sampler type
+             assert.ok(
+                 sampler.startsWith('probability'),
+                 `sampler ${sampler} is not of type 'probability'`);
+
+             // Validate set probability
+             assert.ok(
+                 sampler.endsWith(`(${testProbability})`),
+                 `sampler ${sampler} is not using probability ${
+                     testProbability}`);
+             resolve();
+           }, 15);
```

Comment:
Can we set the server wait time as a top-level constant? Just for ease-of-use (might be useful to tweak if CI flakes occur)